src/runtime/moduleobject.cs

```diff
@@ -341,6 +341,20 @@ protected override void OnSave(InterDomainContext context)
             // Decref twice in tp_clear, equilibrate them.
             Runtime.XIncref(dict);
             Runtime.XIncref(dict);
+            // destroy the cache(s)
+            foreach (var pair in cache)
+            {
+                Runtime.PyDict_DelItemString(dict, pair.Key);
+                pair.Value.DecrRefCount();
+                 if (Exceptions.ExceptionMatches(Exceptions.KeyError))
```

Comment:
weird indent

Reply:
also do you need to PyErr_Clear() before starting the loop? Could someone have set the error for unrelated reasons before entering here?

Reply:
I think it's a non-problem. Either:
* Someone an exception other than KeyError and 
** we never set one and we don't clear or clobber it 
** we clobber it by setting a KeyError
* Someone set a KeyError before us and we clear in the loop (instead of clearing before the loop).

Reply:
You should only make the check when `PyDict_DelItemString` actually returns `-1`